collect_app/src/test/java/org/odk/collect/android/formentry/audit/FormSaveViewModelTest.java

```diff
@@ -68,38 +68,44 @@ public void setup() {
 
     @Test
     public void saveForm_returnsNewSaveResult_inSavingState() {
-        LiveData<FormSaveViewModel.SaveResult> saveResult1 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-        assertThat(saveResult1.getValue().getState(), equalTo(SAVING));
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        FormSaveViewModel.SaveResult saveResult1 = viewModel.getSavedResult().getValue();
+        assertThat(saveResult1.getState(), equalTo(SAVING));
 
-        LiveData<FormSaveViewModel.SaveResult> saveResult2 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        FormSaveViewModel.SaveResult saveResult2 = viewModel.getSavedResult().getValue();
         assertThat(saveResult2, not(equalTo(saveResult1)));
     }
 
     @Test
     public void saveForm_wontRunMultipleSavesAtOnce() {
-        LiveData<FormSaveViewModel.SaveResult> saveResult1 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-        LiveData<FormSaveViewModel.SaveResult> saveResult2 = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
-
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        LiveData<FormSaveViewModel.SaveResult> saveResult1 = viewModel.getSavedResult();
         assertThat(saveResult1.getValue().getState(), equalTo(SAVING));
+
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        LiveData<FormSaveViewModel.SaveResult> saveResult2 = viewModel.getSavedResult();
         assertThat(saveResult2.getValue().getState(), equalTo(ALREADY_SAVING));
+
         assertThat(Robolectric.getBackgroundThreadScheduler().size(), equalTo(1));
 
         whenFormSaverFinishes(SaveFormToDisk.SAVED);
         assertThat(saveResult1.getValue().getState(), equalTo(SAVED));
-        assertThat(saveResult2.getValue().getState(), equalTo(ALREADY_SAVING));
     }
 
     @Test
     public void saveForm_whenReasonRequiredToSave_returnsSaveResult_inChangeReasonRequiredState() {
         whenReasonRequiredToSave();
 
-        LiveData<FormSaveViewModel.SaveResult> saveResult = viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        viewModel.saveForm(Uri.parse("file://form"), true, "", false);
+        LiveData<FormSaveViewModel.SaveResult> saveResult = viewModel.getSavedResult();
```

Comment:
I think having the `saveResult` declarations before the call to `save` would be good because it ensures that the same `LiveData` instance is being used before/after the call. This mimics how you would use the `ViewModel` where you make an observation before you call an action.

Reply:
Done.